distribution/lib/Standard/Test/0.0.0-dev/src/Must_Reach.enso

```diff
@@ -0,0 +1,23 @@
+from Standard.Base import all
+import Standard.Base.Runtime.Ref.Ref
+
+import project.Spec_Result.Spec_Result
+from project import Test
+from project.Extensions import all
+
+type Must_Reach
+    Value (reached_ref : Ref Boolean)
+
+    new : Must_Reach
+    new = Must_Reach.Value (Ref.new False)
+
+    reached : Nothing
+    reached self -> Nothing =
+        self.reached_ref.put True
+        Nothing
+
+    must_have_reached : Integer -> Spec_Result
```

Comment:
why not `should_have_reached`? All of our test suite is built around `should_` so it feels weird to have `must` here unless there's some good reason